src/validation.cpp

```diff
@@ -5543,6 +5548,28 @@ SnapshotCompletionResult ChainstateManager::MaybeCompleteSnapshotValidation(
     return SnapshotCompletionResult::SUCCESS;
 }
 
+Chainstate& ChainstateManager::GetChainstateForNewBlock(const uint256& blockhash)
+{
+    AssertLockHeld(::cs_main);
+    // Early return to avoid unnecessary blockindex lookup when assumeutxo is not in use.
+    if (!m_snapshot_chainstate) {
```

Comment:
In c14ae132c5a5204a9a755c84c6de05fb30459221, is there a reason for not using `IsSnapshotActive()` here?